frame/stake-tracker/migrations/src/lib.rs

```diff
@@ -0,0 +1,235 @@
+pub(crate) const LOG_TARGET: &str = "runtime::stake-tracker";
+
+#[macro_export]
+macro_rules! log {
+	($level:tt, $patter:expr $(, $values:expr)* $(,)?) => {
+		log::$level!(
+			target: crate::LOG_TARGET,
+			concat!("💸🧐", $patter), $(, $values)*
+		)
+	};
+}
+
+pub mod v2 {}
+//TODO: Introduce try-state
+pub mod v1 {
+	use codec::{Decode, Encode};
+	use frame_election_provider_support::SortedListProvider;
+	use frame_support::{
+		pallet_prelude::*, sp_io, storage_alias, traits::OnRuntimeUpgrade, CloneNoBound, EqNoBound,
+		PartialEqNoBound,
+	};
+	use pallet_stake_tracker::{ApprovalStake, BalanceOf, Pallet};
+	use pallet_staking::{Nominations, Nominators, Validators};
+	use sp_runtime::Saturating;
+	use std::collections::BTreeMap;
+
+	/// Migration implementation that injects all validators into sorted list.
+	pub struct InjectValidatorsApprovalStakeIntoTargetList<T>(PhantomData<T>);
+
+	#[derive(Encode, Decode, CloneNoBound, PartialEqNoBound, EqNoBound, Default)]
+	#[codec(mel_bound())]
+	pub struct MigrationState {
+		pub last_key: Vec<u8>,
+		pub prefix: Vec<u8>,
+		pub done: bool,
+	}
+
+	#[storage_alias]
+	type MigrationV1StateNominators<T: pallet_staking::Config + pallet_stake_tracker::Config> =
```

Comment:
Nominators already have their own list in all runtimes, why are re-populating anything here?